### ISSUE TITLE
[generate:entity:content] Default user_id field has unnecessary default value callback

### ISSUE THREAD

**Issue by wizonesolutions** [hechoendrupal/drupal-console#1121]
I see this (I removed the form stuff):

``` php
    $fields['user_id'] = BaseFieldDefinition::create('entity_reference')
      ->setLabel(t('Authored by'))
      ->setDescription(t('The user ID of author of the FillPDF file context entity.'))
      ->setRevisionable(TRUE)
      ->setSetting('target_type', 'user')
      ->setSetting('handler', 'default')
      ->setDefaultValueCallback('Drupal\node\Entity\Node::getCurrentUserId')
      ->setTranslatable(TRUE);
```

But we also have this:

``` php
  /**
   * {@inheritdoc}
   */
  public static function preCreate(EntityStorageInterface $storage_controller, array &$values) {
    parent::preCreate($storage_controller, $values);
    $values += array(
      'user_id' => \Drupal::currentUser()->id(),
    );
  }
```

And it does the same thing `\Drupal\node\Entity\Node::getCurrentUserId()` does anyway.

Using a Node function for this has the consequence of adding a dependency on the Node module. If you must have a default value callback, it's probably better just to generate one, or find a different one that is fit for purpose.

**Comment by tsvenson** [comment]
Just noticed this too. Is the callback needed at all if the proCreate function accomplishes the same thing?

**Comment by HaleFx** [comment]
The callback is not needed

**Comment by miguel303** [comment]
Since the callback is not needed , I removed it from template which generate that class. 
PR in #2440

**Comment by jmolivas** [comment]
Closing this one. It was fixed with #2440
